utils/umaApi.ts

```diff
@@ -144,91 +266,18 @@ type GetStatBetween = (
 ) => Promise<SynthStats[]>;
 
 type GetPriceSlice = (address: string) => Promise<string[]>;
-
-const getAddresses: GetAddresses = async () => {
-  const empAddresses: string[] = await request("listEmpAddresses");
-  const lspAddresses: string[] = await request("listAddresses");
-  return [...empAddresses, ...lspAddresses];
-};
-
-const getState: GetState = async <T extends { type: ContractType }>(
-  address: string
-) => {
-  const state: SynthState<T> = await request("global/getState", address);
-  return state;
-};
-
-const getSynthStats: GetSynthStats = async (address) => {
-  return {
-    id: address,
-    address,
-    tvl: await getLatestTvl(address).catch(() => "0"),
-    tvm: await getLatestTvm(address).catch(() => "0"),
-  };
-};
-
-const time90DaysAgo = nDaysAgo(90);
-const oneDayAgo = nDaysAgo(1);
-
-const getLatestTvl: GetStat = (address) =>
-  address ? request("global/tvl", address) : request("global/globalTvl");
-const getLatestTvm: GetStat = (address) =>
-  address ? request("global/tvm", address) : request("global/globalTvm");
-const getTvl: GetStatBetween = (
-  address,
-  startTimestamp = Math.floor(time90DaysAgo().toSeconds())
-) => request("tvlHistoryBetween", address, startTimestamp);
-
-// FIXME: This the bot price, not the last traded price. Needs to be handled API side.
-const getYesterdayPrice: GetPriceSlice = (address: string) =>
-  request(
-    "sliceHistoricalSynthPrices",
-    address,
-    Math.floor(oneDayAgo().toSeconds())
-  );
-export const client = {
-  request,
-  getAddresses,
-  getState,
-  getSynthStats,
-  getLatestTvl,
-  getLatestTvm,
-  getTvl,
-  getYesterdayPrice,
-};
-
-export async function fetchCompleteSynth<T extends { type: ContractType }>(
+type FetchCompleteSynth = <T extends { type: ContractType }>(
   synth: ContentfulSynth
-): Promise<Synth<T> | Error> {
-  try {
-    const stats = await client.getSynthStats(synth.address);
-    const state = await client.getState<T>(synth.address);
-    const lastTvl = await client.getLatestTvl(synth.address).catch(() => "0");
-    const [{ value: ydayTvl = NaN } = {}] = await client.request(
-      "global/tvlHistorySlice",
-      synth.address,
-      Math.floor(oneDayAgo().toSeconds())
-    );
-    const tvl24hChange = !Number.isNaN(ydayTvl)
-      ? Math.round(
-          ((formatWeiString(lastTvl) - formatWeiString(ydayTvl)) /
-            formatWeiString(ydayTvl)) *
-            1000
-        ) / 10
-      : 0;
-
-    return {
-      ...stats,
-      ...state,
-      tvl24hChange,
-      ...synth,
-    };
-  } catch (err) {
-    return new SynthFetchingError(err.message, synth);
-  }
-}
+) => Promise<Synth<T> | Error>;
 
-// longTokenName can return undefined if value wasn't found in API so we need to do a null check.
-export function formatLSPName(longTokenName: string): string {
-  return longTokenName?.substring(-2);
-}
+type Client = {
```

Comment:
It's just a personal preference, but I'd implement this client as a class, so as types are defined automatically for the class itself and it's functions.

Reply:
agree, typescript is annoyingly biased in how you write your code.  id actually suggest you use the `constructClient` as the factory which takes chain id and maps it to base url.  Then your client is a class that only takes `baseUrl` in the constructor, that way the class isnt dependent on the chainid->url mapping and you can memoize the factory function.

Reply:
not sure about memoization, but it should work, yes. Worst case you'd implement the singleton pattern by yourself for instantiating a single class for each chain